bin/alluxio-start.sh

```diff
@@ -516,6 +516,11 @@ main() {
       start_proxies
       ;;
     local)
+      ALLUXIO_MASTER_JAVA_OPTS+=" -Dalluxio.master.hostname=localhost"
+      ALLUXIO_WORKER_JAVA_OPTS+=" -Dalluxio.master.hostname=localhost"
+      ALLUXIO_PROXY_JAVA_OPTS+=" -Dalluxio.master.hostname=localhost"
+      ALLUXIO_JOB_MASTER_JAVA_OPTS+=" -Dalluxio.master.hostname=localhost"
+      ALLUXIO_JOB_WORKER_JAVA_OPTS+=" -Dalluxio.master.hostname=localhost"
```

Comment:
Is there no common Java opt we can use? Also what will the behavior be if the user does specify a master address?

Reply:
- setting `ALLUXIO_JAVA_OPTS` will not work as `ALLUXIO_MASTER_JAVA_OPTS` will be set in `libexec/config.sh` before reaching here. 
- `alluxio-start.sh local` will overite the master hostname setting in `alluxio-site.properties` which I think should be the correct behavior (when arg `local` is conflicting with parameters in `alluxio-site.properties` )